scene/2d/camera_2d.cpp

```diff
@@ -250,8 +251,10 @@ void Camera2D::_notification(int p_what) {
 			add_to_group(group_name);
 			add_to_group(canvas_group_name);
 
-			if (!Engine::get_singleton()->is_editor_hint() && enabled && !viewport->get_camera_2d()) {
-				make_current();
+			if (!_is_editing_in_editor()) {
+				if (enabled && !viewport->get_camera_2d()) {
```

Comment:
Why did you split the condition? If you think the line is too long you can break it, but the inner condition is redundant.